File: Python/Odious_Number.py
print("Odious Number Checker\n")
n = int(input("Enter the number : "))
flag = False
m = n
nfo = 0
if n <  0:
	n *= -1
while n>1 :
	nfo += n%2
	n //= 2
if n==1:
	nfo += 1
if nfo%2!=0 and m>0:
	print(f"{m} is an odious number.")
else:
	print(f"{m} is not an odious number.")

File: Java/src/Odious_Number.java
/**
 * Odious Number Checker
 * 
 * @author Jaysmito Mukherjee 
 * @version 2019-A-022
 */
import java.util.*;
public class Odious_Number
{
    public static void main()
    {
        System.out.print("\f");
        Scanner sc=new Scanner(System.in);
        System.out.print("Odious Number Checker\n~~~~~~ ~~~~~~ ~~~~~~~\nEnter the number: ");
        int n=sc.nextInt();
        int m=n,c=0,t;
        while(m>0)
        {
            t=m%2;
            m=m/2;
            if(t==1)
                c++;
        }
        if(c%2!=0)
            System.out.print(n+" is an Odious number.");
        else
            System.out.print(n+" is not an Odious number.");        
    }
}